packages/core/src/html/NodeProcessor.ts

```diff
@@ -116,34 +133,41 @@ class NodeProcessor {
    * Layout element collection
    */
 
-  _collectLayoutEl(node, property) {
+  private static collectLayoutEl(node: DomElement): string | null {
     const $ = cheerio(node);
-    this[property].push($.html());
+    const html = $.html();
     $.remove();
+    return html;
   }
 
   /**
    * Removes the node if modal id already exists, processes node otherwise
    */
-  _processModal(node) {
-    if (this.processedModals[node.attribs.id]) {
-      cheerio(node).remove();
-    } else {
-      this.processedModals[node.attribs.id] = true;
+  private processModal(node: DomElement) {
+    if (node.attribs) {
```

Comment:
only change here is this. This lets tsc know `node.attribs` (which is defined as `?: {}`) always exists